platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/TypedResolveException.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.artifacts.ivyservice;
+
+import com.google.common.collect.ImmutableList;
+import org.gradle.api.artifacts.ResolveException;
+
+import java.util.List;
+
+/**
+ * An internal specialization of the public {@link ResolveException}. All resolve exceptions thrown
+ * by Gradle are assumed to be an instance of this class.
+ */
+public class TypedResolveException extends ResolveException {
+    private final String type;
```

Comment:
I'm not really sure I see the value in converting to an enum here. To me that's just added complexity without any significant advantage

Reply:
The advantage I was thinking of was that the reader of this exception code could see the range of possible "types" it works with at a glance.

Reply:
But changing those types doesn't actually affect any behavior. The exception code works with _all_ types. If there was a distinction between the types and logic changed depending on the type I could see value in making the possibilities more strict, but this data is only used for the exception message.

Reply:
Right, it's not about changing the behavior, it's about making the source code more comprehensible for us, instead of having to search for the different types that could be supplied here.

Reply:
But I'm saying the type used here never really matters. I don't see a reason someone would want to search for the types used here

Reply:
> I don't see a reason someone would want to search for the types used here

I just did when reading the review. :-)

I was thinking when grepping for error messages.  It would make it easier to tell where this exception came from.